### ISSUE TITLE
NA values break plot.errors

### ISSUE THREAD

**Issue by richardsc** [r-quantities/errors#35]
Just discovered this package (after thinking about writing my own) -- it's great! Thanks!

I noticed that `xlim` and `ylim` are hard-coded in the `NextMethod()` call for `plot.errors()` to be `range(x)` and `range(y)`.

One problem with this is that it is impossible to control the axis limits of the plot (say, to focus on a region or reverse an axes).

Another issue is that if `x` or `y` have NAs they cause `plot.errors()` to fail:

```r
library(errors)
x <- 1:100
y <- rnorm(x)
y[20] <- NA
errors(x) <- 0
errors(y) <- rnorm(y, sd=0.25)
plot(x, y)
```
```
Error in plot.window(...) : need finite 'ylim' values
```

I don't see why `plot.errors()` isn't simply written as:
```r
plot.errors <- function(x, y, ...) {
  if (missing(y)) {
    y <- x
    x <- seq_along(x)
  }

  NextMethod()

  if (inherits(x, "errors"))
    graphics::segments(errors_min(x), y, errors_max(x), y)

  if (inherits(y, "errors"))
    graphics::segments(x, errors_min(y), x, errors_max(y))
}
```

Note that using the `na.rm=TRUE` argument in `range()` doesn't help, as it still returns `c(NA, NA)` for `range(y)`. I assume that this is due to the interaction the base `range()` function with the `errors-class` object and it doesn't know what to do with it?

**Comment by dankelley** [comment]
I also have come to the errors package only in the past few weeks (and I had actually written a similar, but much less useful package, a few days ago).

This `range` problem is in `R/summary.R` line 32, viz.
```R
    "range" = c(min(...), max(...))
```
which I think ought to read
```R
    "range" = c(min(..., na.rm=na.rm), max(..., na.rm=na.rm))
```

**Comment by dankelley** [comment]
Oh, and also the `which.min` and `which.max` calls above line 32 could also use adjustment.

**Comment by Enchufa2** [comment]
Thanks for your reports.

@dankelley You are right about `range`, it should be fixed in 8f95673. `which.min` and `which.max` have no `na.rm` argument though.

@richardsc

> I don't see why plot.errors() isn't simply written as [...]

 The problem is that we don't want error bars to be cut out of the plotting range by default, but I agree that a user-provided `xlim` or `ylim` should be respected. So it's a little more complicated than that. I'm working on it.

**Comment by Enchufa2** [comment]
In fact, it was a incorrect implementation of `min`, `max` and `range`. Rewritten in 61b004c.

**Comment by Enchufa2** [comment]
`plot` fixed in a3a6f0f, I think. Could you please test the current master branch, @richardsc?
